### PR TITLE
Refactor iterator-related tests

### PR DESCRIPTION
**Pull request by mina86**
Firstly, introduce helper assert_iter and assert_iter_reversed
functions to reduce code repetition.  This will be especially useful
in upcoming commit which changes the Item type of the iterator.

Secondly, keep keys and values declared at the start of some of the
tests as slices without boxing them.  Doing the latter introduces
conveluted syntax such as `&*k1`; keeping them as slices simplifies
the syntax and also allows them to be converte into constsants.

Lastly, replace `cba` helper function with `pair` which construct
a pair of boxed slices.  This removes some repetition since after
the previous point `cba` was only ever used in a pair.

### CODE REVIEW

File: tests/util/mod.rs [hunk 1]
```diff
@@ -38,3 +38,26 @@ impl AsRef<Path> for &DBPath {
...
+
+#[allow(dead_code)]
+pub fn pair(left: &[u8], right: &[u8]) -> (Box<[u8]>, Box<[u8]>) {
+    (Box::from(left), Box::from(right))
+}
+
+// Use macro rather than functions so that on failure we get line number where
+// the macro was instantiated rather than line number inside of a function.
+#[allow(unused_macros)]
+macro_rules! assert_iter {
```
Comment (stanislav-tkach) [inline tests/util/mod.rs:50]: Can this macro be replaced with a function with the `#[track_caller]` attribute?

**Review by stanislav-tkach — APPROVED** [review]

File: tests/util/mod.rs (hunk 1 above)
Comment (mina86) [reply tests/util/mod.rs]: Perfect. Done.

**Comment by mina86** [comment]
Friendly ping.

**Comment by mina86** [comment]
Sorry, didn’t notice I’d broken clippy. Fixed now.

**Review by aleksuss — APPROVED** [review]

**Pull request merged**